addons/mail/static/src/models/thread_view/thread_viewer.js

```diff
@@ -124,7 +124,11 @@ function factory(dependencies) {
             if (!this.thread) {
                 return [['unlink']];
             }
-            return [['link', this.thread.cache(this.stringifiedDomain)]];
+            const cache = this.thread.cache(this.stringifiedDomain);
+            if (cache != this.threadCache) {
+                cache.thread.markMessagesAsOriginThreadAsRead();
+            }
```

Comment:
Issue likely comes from calling this before the thread cache is loaded, and with bad luck the return from message_fetch is resetting the message as a needaction in JS in parallel of the other one setting it to false.